sentry-android-core/src/main/java/io/sentry/android/core/SentryPerformanceProvider.java

```diff
@@ -92,62 +56,146 @@ public String getType(@NotNull Uri uri) {
+      app = (Application) context;
+    }
+    if (app == null) {
+      return;
     }
+
+    final @NotNull TimeSpan appStartTimespan = appStartMetrics.getAppStartTimeSpan();
+    appStartTimespan.setStartedAt(Process.getStartUptimeMillis());
+
+    final AtomicBoolean firstDrawDone = new AtomicBoolean(false);
+    final Handler handler = new Handler(Looper.getMainLooper());
+
+    activityCallback =
+        new ActivityLifecycleCallbacksAdapter() {
+          final WeakHashMap<Activity, ActivityLifecycleTimeSpan> activityLifecycleMap =
+              new WeakHashMap<>();
+
+          @Override
+          public void onActivityPreCreated(
+              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+            final long now = SystemClock.uptimeMillis();
+            if (appStartMetrics.getAppStartTimeSpan().hasStopped()) {
+              return;
+            }
+
+            final ActivityLifecycleTimeSpan timeSpan = new ActivityLifecycleTimeSpan();
+            timeSpan.getOnCreate().setStartedAt(now);
+            activityLifecycleMap.put(activity, timeSpan);
+          }
+
+          @Override
+          public void onActivityCreated(
+              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+            if (appStartMetrics.getAppStartType() == AppStartMetrics.AppStartType.UNKNOWN) {
+              appStartMetrics.setAppStartType(
+                  savedInstanceState == null
+                      ? AppStartMetrics.AppStartType.COLD
+                      : AppStartMetrics.AppStartType.WARM);
+            }
+          }
+
+          @Override
+          public void onActivityPostCreated(
+              @NonNull Activity activity, @Nullable Bundle savedInstanceState) {
+            if (appStartMetrics.getAppStartTimeSpan().hasStopped()) {
+              return;
+            }
+
+            final @Nullable ActivityLifecycleTimeSpan timeSpan = activityLifecycleMap.get(activity);
+            if (timeSpan != null) {
+              timeSpan.getOnCreate().stop();
+              timeSpan.getOnCreate().setDescription(activity.getClass().getName() + ".onCreate");
+            }
+          }
+
+          @Override
+          public void onActivityPreStarted(@NonNull Activity activity) {
+            final long now = SystemClock.uptimeMillis();
+            if (appStartMetrics.getAppStartTimeSpan().hasStopped()) {
+              return;
+            }
+            final @Nullable ActivityLifecycleTimeSpan timeSpan = activityLifecycleMap.get(activity);
+            if (timeSpan != null) {
+              timeSpan.getOnStart().setStartedAt(now);
+            }
+          }
+
+          @Override
+          public void onActivityStarted(@NonNull Activity activity) {
+            if (firstDrawDone.get()) {
+              return;
+            }
+            FirstDrawDoneListener.registerForNextDraw(
+                activity,
+                () ->
+                    handler.postAtFrontOfQueue(
```

Comment:
this `postAtFrontOfQueue` is already taken care of in the [FirstDrawDoneListener](https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/internal/util/FirstDrawDoneListener.java#L84)